### PR TITLE
build-patched-kao-kcmo-images.sh: get image name from brew container image metadata

### PR DESCRIPTION
**Pull request by redbeam**
#871

### CODE REVIEW

**Review by praveenkumar — CHANGES_REQUESTED** [review]

**Comment by praveenkumar** [comment]
@redbeam both commit  can be squashed to have a single one.

File: build-patched-kao-kcmo-images.sh [hunk 1]
```diff
@@ -99,9 +99,9 @@ function patch_and_push_image() {
         rhpkg container-build --target crc-1-rhel-9-candidate
         popd
     fi
-    # Operator images created using rhel-9 tags have `rhel9-operator` as part of image name so replacing `operator` with it.
-    # https://www.gnu.org/software/bash/manual/bash.html#Shell-Parameter-Expansion
-    rhel9_image_name="${image_name/operator/rhel9-operator}"
+    # Metadata output from pull list offers 2 variants - SHA digest variant and tag variant.
+    # They are essentially the same, so we can choose either ([0] or [1]).
+    rhel9_image_name=$(curl -L https://download.eng.bos.redhat.com/brewroot/packages/crc-cluster-kube-apiserver-operator-container/${version}/${release}/metadata.json | jq -r '.build.extra.image.index.pull[0]')
```
[the remark this answers is not in the document]
Comment (redbeam) [reply build-patched-kao-kcmo-images.sh]: Yes, I was unsure about the variable name, so I just left it as it was.

I made the changes you suggested.

**Comment by redbeam** [comment]
@praveenkumar Squashed.

**Pull request merged**